telluric/georaster.py

```diff
@@ -1518,38 +1494,18 @@ def _get_widow_calculate_resize_ratio(self, xsize, ysize, window):
 
         return xratio, yratio
 
-    def _get_window_out_shape(self, bands, xratio, yratio, window):
+    def _get_window_out_shape(self, bands, window, xsize, ysize):
```

Comment:
now that `xratio` and `yratio` are not needed inside `get_window` you can modify the method, and combine `_get_window_out_shape` and `_get_window_calculate_resize_ratio` to more efficient code, and not do the calculations and rounding when `bands, xsize, ysize` are all presented:
```
    def _get_window_out_shape(self, bands, xsize, ysize, window):
        """Get the outshape of a window.

        this method is only used inside get_window to calculate the out_shape
        """
        if xsize and ysize:
            out_shape = (len(bands), ysize, xsize)
        elif xsize and ysize is None:
            ratio = math.ceil(abs(window.width) / xsize)
            out_shape = (len(bands), math.ceil(abs(window.height) / ratio), xsize)
        elif ysize and xsize is None:
            ratio = math.ceil(abs(window.height) / ysize)
            out_shape = (len(bands), ysize, math.ceil(abs(window.width) / ratio))
        else:
            out_shape = (len(bands), math.ceil(abs(window.height)), math.ceil(abs(window.width)))
        return out_shape
```

Reply:
that it is not the point, the problem is that in the two methods  way you are always return this `out_shape = (len(bands), math.ceil(abs(window.height)), math.ceil(abs(window.width)))` even like in most cases when you have this ` (len(bands), ysize, xsize)`.
which can result in rounding issues